Packs/Bitbucket/Integrations/Bitbucket/Bitbucket.py

```diff
@@ -0,0 +1,1481 @@
+    )
+    return CommandResults(
+        readable_output=readable_output,
+        outputs_prefix='Bitbucket.Branch',
+        outputs=response,
+        raw_response=response
+    )
+
+
+def branch_create_command(client: Client, args: Dict) -> CommandResults:
+    """ creates a branch in Bitbucket.
+    Args:
+        client: A Bitbucket client.
+        args: Demisto args.
+    Returns:
+        A CommandResult object with a success message, in case of a successful action.
+    """
+    repo = args.get('repo')
+    name = args.get('name', '')
+    target_branch = args.get('target_branch', '')
+    if not repo:
+        repo = client.repository
+    response = client.branch_create_request(name, target_branch, repo)
+    return CommandResults(
+        readable_output=f'The branch "{name}" was created successfully.',
+        outputs_prefix='Bitbucket.Branch',
+        outputs=response,
+        raw_response=response
+    )
+
+
+def branch_delete_command(client: Client, args: Dict) -> CommandResults:
+    """ Deletes a branch in Bitbucket.
+    Args:
+        client: A Bitbucket client.
+        args: Demisto args.
+    Returns:
+        A CommandResult object with a success message, in case of a successful action.
+    """
+    repo = args.get('repo')
+    branch_name = args.get('branch_name', '')
+    if not repo:
+        repo = client.repository
+    response = client.branch_delete_request(branch_name, repo)
+    if response.status_code == 204:
+        return CommandResults(readable_output=f'The branch {branch_name} was deleted successfully.')
+    else:
+        raise Exception('The command branch-delete failed.')
+
+
+def commit_create_command(client: Client, args: Dict) -> CommandResults:
+    """ creates a commit in Bitbucket.
+    Args:
+        client: A Bitbucket client.
+        args: Demisto args.
+    Returns:
+        A CommandResult object with a success message, in case of a successful action.
+    """
+    repo = args.get('repo')
+    message = args.get('message', '')
+    branch = args.get('branch', '')
+    file_name = args.get('file_name', None)
+    file_content = args.get('file_content', None)
+    entry_id = args.get('entry_id', None)
+    author_name = args.get('author_name', None)
+    author_email = args.get('author_email', None)
+
+    if not file_name and not entry_id:
+        raise Exception('You must specify either the "file_name" and "file_content" or the "entry_id" of the file.')
+    elif file_name and entry_id:
+        raise Exception('You must specify the "file_name" and "file_content" or the "entry_id" of the file, not both.')
+    elif entry_id:
+        file_path = demisto.getFilePath(entry_id).get('path')
+        file_name = demisto.getFilePath(entry_id).get('name')
+        with open(file_path, 'rb') as f:
+            file_content = f.read()
+    body = {
+        "message": message,
+        "branch": branch,
+        file_name: file_content
+    }
+    if author_name and author_email:
```

Comment:
If only one of them is given? what's the behavior then?

Reply:
The behavior here is simply not to add this information to the body, and send it to the api without this information (and use the default author which is the user name and password that was entered in order to connect to Bitbucket.
In a second thought, I added now an exception that will tell the user to give the missing detail as well.